src/Icon.php

```diff
@@ -0,0 +1,142 @@
+<?php
+/**
+ * Class Icon.
+ *
+ * @package AmpProject\AmpWP
+ */
+
+namespace AmpProject\AmpWP;
+
+/**
+ * Icons used to visually represent the state of a validation error.
+ *
+ * @package AmpProject\AmpWP
+ */
+class Icon {
```

Comment:
Class should be `final` by default:
```suggestion
final class Icon {
```

Otherwise, you'd need to change the calls to `self::` to `static::` to use late static binding so an extending class would not unexpectedly break.

This would mean we advertise this class for extension, though, and we should only do that if the actual benefit outweighs the additional maintenance cost.